be/src/http/action/stream_load_with_sql.cpp

```diff
@@ -0,0 +1,417 @@
+        // RETURN_IF_ERROR(_exec_env->stream_load_executor()->commit_txn(ctx));
+        ctx->commit_and_publish_txn_cost_nanos = MonotonicNanos() - commit_and_publish_start_time;
+    }
+    while (!ctx->is_stream_load_put_success) {
+    }
+    return ctx->status;
+}
+
+int StreamLoadWithSqlAction::on_header(HttpRequest* req) {
+    streaming_load_with_sql_current_processing->increment(1);
+
+    StreamLoadContext* ctx = new StreamLoadContext(_exec_env);
+    ctx->ref();
+    req->set_handler_ctx(ctx);
+
+    ctx->load_type = TLoadType::MANUL_LOAD;
+    ctx->load_src_type = TLoadSourceType::RAW;
+
+    ctx->db = req->param(HTTP_DB_KEY);
+    ctx->table = req->param(HTTP_TABLE_KEY);
+    ctx->label = req->header(HTTP_LABEL_KEY);
+    if (ctx->label.empty()) {
+        ctx->label = generate_uuid_string();
+    }
+
+    ctx->two_phase_commit = req->header(HTTP_TWO_PHASE_COMMIT) == "true" ? true : false;
+
+    LOG(INFO) << "new income streaming load request." << ctx->brief() << ", db=" << ctx->db
+              << ", tbl=" << ctx->table;
+
+    auto st = _on_header(req, ctx);
+    if (!st.ok()) {
+        ctx->status = std::move(st);
+        if (ctx->need_rollback) {
+            _exec_env->stream_load_executor()->rollback_txn(ctx);
+            ctx->need_rollback = false;
+        }
+        if (ctx->body_sink.get() != nullptr) {
+            ctx->body_sink->cancel(ctx->status.to_string());
+        }
+        auto str = ctx->to_json();
+        // add new line at end
+        str = str + '\n';
+        HttpChannel::send_reply(req, str);
+        streaming_load_with_sql_current_processing->increment(-1);
+#ifndef BE_TEST
+        if (config::enable_stream_load_record) {
+            str = ctx->prepare_stream_load_record(str);
+            _save_stream_load_record(ctx, str);
+        }
+#endif
+        return -1;
+    }
+    return 0;
+}
+
+Status StreamLoadWithSqlAction::_on_header(HttpRequest* http_req, StreamLoadContext* ctx) {
+    // auth information
+    if (!parse_basic_auth(*http_req, &ctx->auth)) {
+        LOG(WARNING) << "parse basic authorization failed." << ctx->brief();
+        return Status::InternalError("no valid Basic authorization");
+    }
+    // default csv
+    ctx->format = TFileFormatType::FORMAT_CSV_PLAIN;
+
+    if (ctx->format == TFileFormatType::FORMAT_UNKNOWN) {
+        return Status::InternalError("unknown data format, format={}",
+                                     http_req->header(HTTP_FORMAT_KEY));
+    }
+
+    // check content length
+    ctx->body_bytes = 0;
+    size_t csv_max_body_bytes = config::streaming_load_max_mb * 1024 * 1024;
+
+#ifndef BE_TEST
+    evhttp_connection_set_max_body_size(
+            evhttp_request_get_connection(http_req->get_evhttp_request()), csv_max_body_bytes);
+#endif
+
+    // begin transaction
+    int64_t begin_txn_start_time = MonotonicNanos();
+    // RETURN_IF_ERROR(_exec_env->stream_load_executor()->begin_txn(ctx));
```

Comment:
begin txn not called? how to set txn_id?

Reply:
> 
begin_txn  on fe